command/genesis/polybft_params.go

```diff
@@ -289,6 +294,61 @@ func (p *genesisParams) deployContracts(totalStake *big.Int) (map[types.Address]
 			artifact: contractsapi.L2StateSender,
 			address:  contracts.L2StateSenderContract,
 		},
+		{
+			artifact: contractsapi.ValidatorSet,
+			address:  contracts.NewValidatorSetContract,
+			constructorCallback: func(artifact *artifact.Artifact,
+				polybftConfig *polybft.PolyBFTConfig,
+				rootConfig *polybft.RootchainConfig) ([]byte, error) {
+				// TODO @goran-ethernal- this will be changed in the next PRs to use generated binding
+				// once we remove the old ChildValidatorSet on use the new ValidatorInit binding
+				validatorsMap := make([]map[string]interface{}, len(polybftConfig.InitialValidatorSet))
+				for i, validator := range polybftConfig.InitialValidatorSet {
+					validatorsMap[i] = map[string]interface{}{
+						"addr":  validator.Address,
+						"stake": validator.Stake,
+					}
+				}
+
+				// TODO @goran-ethernal - we will remove once we change e2e tests
+				// since by RFC-201 we won't be able to start edge without bridge (root)
+				customSupernetManagerAddr := types.ZeroAddress
+				if rootConfig != nil {
+					customSupernetManagerAddr = rootConfig.CustomSupernetManagerAddress
+				}
+
+				encoded, err := artifact.Abi.Constructor.Inputs.Encode([]interface{}{
+					contracts.L2StateSenderContract,
+					contracts.StateReceiverContract,
+					customSupernetManagerAddr,
+					new(big.Int).SetUint64(polybftConfig.EpochSize),
+					validatorsMap,
+				})
+				if err != nil {
+					return nil, err
+				}
+
+				return append(artifact.Bytecode, encoded...), nil
+			},
+		},
+		{
+			artifact: contractsapi.RewardDistributor,
+			address:  contracts.RewardDistributorContract,
+			constructorCallback: func(artifact *artifact.Artifact,
+				polybftConfig *polybft.PolyBFTConfig,
+				rootConfig *polybft.RootchainConfig) ([]byte, error) {
+				encoded, err := artifact.Abi.Constructor.Inputs.Encode([]interface{}{
+					contracts.NativeERC20TokenContract, // TODO @goran-ethernal - we will use a different token for rewards
```

Comment:
Just to add my two cents to TODO:
maybe we should consider to parameterize whether we are using native token for rewards or different one. However it is a question for product.

Reply:
Agreed.